examples/convert_empty_room.py

```diff
@@ -0,0 +1,91 @@
+"""
+.. _ex-convert-mne-sample:
+
+======================================
+Storing empty room data in BIDS format
+======================================
+
+This example demonstrates how to store empty room data in BIDS format
+and how to retrieve them.
+"""
+
+# Authors: Mainak Jas <mainakjas@gmail.com>
+#
+# License: BSD (3-clause)
+
+###############################################################################
+# Let us first import mne_bids
+
+import os.path as op
+
+from datetime import datetime
+
+import mne
+from mne.datasets import sample
+
+from mne_bids import write_raw_bids, read_raw_bids, make_bids_basename
+
+###############################################################################
+# And define the paths and event_id dictionary.
+
+data_path = sample.data_path()
+output_path = op.join(data_path, '..', 'MNE-sample-data-bids')
+raw_fname = op.join(data_path, 'MEG', 'sample', 'sample_audvis_raw.fif')
+
+# Specify the raw_file and events_data and run the BIDS conversion.
+raw = mne.io.read_raw_fif(raw_fname)
+bids_basename = make_bids_basename(subject='01', session='01',
+                                   task='audiovisual', run='01')
+write_raw_bids(raw, bids_basename, output_path, overwrite=True)
+
+###############################################################################
+# Specify some empty room data and run BIDS conversion on it.
+er_raw_fname = op.join(data_path, 'MEG', 'sample', 'ernoise_raw.fif')
+er_raw = mne.io.read_raw_fif(er_raw_fname)
+
+# For empty room data we need to specify the recording date in the format
+# YYYYMMDD for the session id.
+er_date = datetime.fromtimestamp(
+    er_raw.info['meas_date'][0]).strftime('%Y%m%d')
+print(er_date)
+
+###############################################################################
+# The measurement date is
+raw_date = datetime.fromtimestamp(
+    raw.info['meas_date'][0]).strftime('%Y%m%d')
+print(raw_date)
+
+###############################################################################
+# We also need to specify that the subject ID is 'emptyroom', and that the 
+# task is 'noise'.
+er_bids_basename = 'sub-emptyroom_ses-{0}_task-noise'.format(er_date)
```

Comment:
make_bids_basename should be able to do this automatically when task=noise and subject=empyroom

so you would not have to do the kung fu with dates above.

Reply:
are you proposing that `make_bids_basename` take `raw` as an input?

Reply:
I think the kung fu is much less now that MNE uses `datetime` objects.

Reply:
it feels wrong in terms of design. make_bids_basename should take care of the formatting of bids
filenames. Here we do  the job of this function. How about you use:

make_bids_basename here but enforce that session is a well formatted date if task is called noise?